cirq/value/value_equality.py

```diff
@@ -0,0 +1,120 @@
+# Copyright 2018 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Union, Callable, overload, Any
+
+from typing_extensions import Protocol
+
+
+class _SupportsValueEquality(Protocol):
+    """An object decorated with the value equality decorator."""
+
+    def _value_equality_values_(self) -> Any:
```

Comment:
I think these need doc strings because this is a protocol and it's not obvious what they are until you read way down lower.

Reply:
Done.